src/Database/Driver.php

```diff
@@ -215,6 +225,73 @@ public function exec(string $sql): int|false
      */
     abstract public function enabled(): bool;
 
+    /**
+     * @inheritDoc
+     */
+    public function execute(string $sql, array $params = [], array $types = []): StatementInterface
+    {
+        $statement = $this->prepare($sql);
+        if (!empty($params)) {
+            $statement->bind($params, $types);
+        }
+        $this->executeStatement($statement);
+
+        return $statement;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function run(Query $query): StatementInterface
+    {
+        $statement = $this->prepare($query);
+        $query->getValueBinder()->attachTo($statement);
+        $this->executeStatement($statement);
+
+        return $statement;
+    }
+
+    /**
+     * Execute the statement and log the query string.
+     *
+     * @param \Cake\Database\StatementInterface $statement Statement to execute.
+     * @return void
+     */
+    protected function executeStatement(StatementInterface $statement): void
+    {
+        if ($this->logger === null) {
+            $statement->execute();
+
+            return;
+        }
+
+        $exception = null;
+        $took = 0.0;
+
+        try {
+            $start = microtime(true);
+            $statement->execute();
+            $took = (float)number_format((microtime(true) - $start) * 1000, 1);
+        } catch (PDOException $e) {
+            $exception = $e;
+        }
+
+        $logContext = [
+            'driver' => $this,
+            'error' => $exception,
+            'params' => $statement->getBoundParams(),
```

Comment:
This doesn't log params passed to Statement::execute().

Reply:
So you want to drop passing params to execute?

Reply:
I'll just add params argument to executeStatement() for now.